semgrep-core/src/parsing/pfff/scala_to_generic.ml

```diff
@@ -121,10 +121,19 @@ let v_dotted_name_of_stable_id (v1, v2) =
   let id = id_of_simple_ref v1 in
   id :: v2
 
-let rec v_import_expr tk (v1, v2) =
-  let module_name = G.DottedName (v_dotted_name_of_stable_id v1) in
-  let v2 = v_import_spec v2 in
-  v2 tk module_name
+let rec v_import_expr tk import_expr =
+  match import_expr with
+  | Left id ->
+      [
+        {
+          G.d = G.ImportAll (tk, FileName id, Parse_info.fake_info tk "");
```

Comment:
I think this should probably be `DottedName` not `FileName`. The case below always uses `DottedName` and I bet the discrepancy here will lead to a failure to match. Sure, we can now parse `import $X` as a pattern, but that doesn't help much if it never matches any source. We can have a `DottedIdent` that contains only a single identifier.

Reply:
You might also run into issues with using `ImportAll` exclusively here, since `v_import_spec` chooses between `ImportFrom` and `ImportAll` when converting to the generic AST...